apps/src/assetManagement/animationLibraryApi.js

```diff
@@ -90,6 +99,113 @@ export function createDefaultSpriteMetadata(listData) {
   });
 }
 
+export function buildAnimationMetadata(files) {
+  let animationMetadataByName = {};
+  let resolvedPromisesArray = [];
+  for (const [fileKey, fileObject] of Object.entries(files)) {
+    let json = fileObject['json'];
+    let png = fileObject['png'];
+    resolvedPromisesArray.push(
+      getAnimationLibraryFile(json.key)
+        .then(metadata => {
+          // Metadata contains name, frameCount, frameSize, looping, frameDelay
+          let combinedMetadata = metadata;
+          combinedMetadata['jsonLastModified'] = json.last_modified;
+          combinedMetadata['pngLastModified'] = png.last_modified;
+          combinedMetadata['version'] = png.version_id;
+          combinedMetadata[
+            'sourceUrl'
+          ] = `/api/v1/animation-library/level_animations/${png.version_id}/${
+            metadata.name
+          }.png`;
+          combinedMetadata['sourceSize'] = png.source_size;
+          animationMetadataByName[fileKey] = combinedMetadata;
+          return Promise.resolve();
+        })
+        .catch(err => {
+          return Promise.reject(err);
+        })
+    );
+  }
+  return Promise.all(resolvedPromisesArray).then(() => {
+    return animationMetadataByName;
+  });
+}
+
+export function buildAliasMap(animationMetadata) {
+  let aliasMap = {};
+
+  for (const [key, metadata] of Object.entries(animationMetadata)) {
+    let aliases = [metadata['name'].toLowerCase()];
+    if (metadata['aliases']) {
+      metadata['aliases'].map(alias => {
+        aliases.push(alias.toLowerCase());
+      });
+    }
```

Comment:
we should use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) for readability

Reply:
along with optional chaining, i think there's a more idiomatic ES6 way we can express this:

```javascript
const formattedAliases = metadata.aliases?.map(alias => alias.toLowerCase());
aliases = [...aliases, ...formattedAliases];
```